tests/test_validation_passive_context_budget.py

```
@@ -0,0 +1,369 @@
+        parser = build_parser()
+        args = parser.parse_args(["--format", "json"])
+        assert args.output_format == "json"
+
+    def test_budget_overrides(self) -> None:
+        parser = build_parser()
+        args = parser.parse_args(["--budget", "a.md:1000", "--budget", "b.md:2000"])
+        assert len(args.budget) == 2
+
+    def test_custom_path(self) -> None:
+        parser = build_parser()
+        args = parser.parse_args(["--path", "/tmp/test"])
```

Ruff (0.15.9)

[error] Probable insecure usage of temporary file or directory: "/tmp/test"

(S108)